src/network.tf

```
@@ -27,3 +27,204 @@ module "vnet_integration" {
+locals {
+  backend_address_pool_name       = format("%s-appgw-be-address-pool", local.project)
+  frontend_http_port_name         = format("%s-appgw-fe-http-port", local.project)
+  frontend_https_port_name        = format("%s-appgw-fe-https-port", local.project)
+  frontend_ip_configuration_name  = format("%s-appgw-fe-ip-configuration", local.project)
+  http_setting_name               = format("%s-appgw-be-http-settings", local.project)
+  http_listener_name              = format("%s-appgw-fe-http-settings", local.project)
+  https_listener_name             = format("%s-appgw-fe-https-settings", local.project)
+  http_request_routing_rule_name  = format("%s-appgw-http-reqs-routing-rule", local.project)
+  https_request_routing_rule_name = format("%s-appgw-https-reqs-routing-rule", local.project)
+  acme_le_ssl_cert_name           = format("%s-appgw-acme-le-ssl-cert", local.project)
+  http_to_https_redirect_rule     = format("%s-appgw-http-to-https-redirect-rule", local.project)
+}
+
+# Application gateway: Multilistener configuraiton
+module "app_gw" {
+  source = "./modules/app_gw"
+
+  resource_group_name = azurerm_resource_group.rg_vnet.name
+  location            = azurerm_resource_group.rg_vnet.location
+  name                = format("%s-app-gw", local.project)
+
+  # SKU
+  sku_name = "WAF_v2"
+  sku_tier = "WAF_v2"
+
+  # Networking
+  subnet_id    = module.appgateway-snet.id
+  public_ip_id = azurerm_public_ip.apigateway_public_ip.id
+
+  # Configure backends
+  backends = {
+    apim = {
+      protocol     = "Https"
+      host         = trim(azurerm_dns_a_record.dns_a_api.fqdn, ".")
+      port         = 443
+      ip_addresses = module.apim.private_ip_addresses
+      probe        = "/status-0123456789abcdef"
+      probe_name   = "probe-apim"
+    }
+
+    portal = {
+      protocol     = "Https"
+      host         = trim(azurerm_dns_a_record.dns_a_portal.fqdn, ".")
+      port         = 443
+      ip_addresses = module.apim.private_ip_addresses
+      probe        = "/signin"
+      probe_name   = "probe-portal"
+    }
+
+    management = {
+      protocol = "Https"
+      #TODO : once the migration is completed there wouldn't be need of the condition below.
+      #       Only the first dns recors is required.
+      host         = var.env_short != "p" ? trim(azurerm_dns_a_record.dns-a-management[0].fqdn, ".") : trim(azurerm_dns_a_record.dns-a-management[0].fqdn, ".")
+      port         = 443
+      ip_addresses = module.apim.private_ip_addresses
+      probe        = "/ServiceStatus"
+      probe_name   = "probe-management"
+    }
+  }
+
+  ssl_profiles = [{
+    name                             = format("%s-issuer-mauth-profile", local.project)
+    trusted_client_certificate_names = null # [format("%s-issuer-chain", local.project)]
+    verify_client_cert_issuer_dn     = true
+    ssl_policy = {
+      disabled_protocols = []
+      policy_type        = "Predefined"
```

```suggestion
      policy_type        = "Custom"
```
inspired by https://github.com/pagopa/io-infrastructure-modules-new/blob/master/azurerm_application_gateway/main.tf

---

@pp-ps can u check if ssl policy is right with this limited cipher_suites?